frontend/src/util/PrivateRoute.tsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { Route, Redirect } from "react-router-dom";
+
+function PrivateRoute({ component: Component, user, ...rest }: any): any {
```

Comment:
Can you make sure it's possible to directly visit a `PrivateRoute` from a new page load?  In Bolt, the reason `PrivateRoute` has a loading state is because when you first load the page, it has to make an ajax request to get information about the signed in user from the server).  Otherwise, `PrivateRoute`s would immediately redirect to the homepage when trying to directly access the page and signed in as an admin.

Reply:
Yeah, look at App.tsx, it's setup so that it makes the graphql request, and displays a spinner when it's loading. Only when the data is retrieved does it decide to reroute or not. Hooks have really come a long way haha. I just tried loading localhost:3000/admin directly and it works fine no issues.